aviary/mission/gasp_based/ode/landing_eom.py

```diff
@@ -11,13 +11,22 @@ class LandingAltitudeComponent(om.ExplicitComponent):
     Compute the landing altitude.
     """
 
+    def initialize(self):
+        self.options.declare('num_nodes', default=1, types=int)
+
     def setup(self):
-        add_aviary_input(self, Mission.Landing.OBSTACLE_HEIGHT, val=50.0)
-        add_aviary_input(self, Mission.Landing.AIRPORT_ALTITUDE, val=0.0)
+        nn = self.options['num_nodes']
+
+        add_aviary_input(self, Mission.Landing.OBSTACLE_HEIGHT, val=np.ones(nn)*50.0)
+        add_aviary_input(self, Mission.Landing.AIRPORT_ALTITUDE, val=np.zeros(nn))
```

Comment:
When we vectorize a group of components for use in the ODE, we need to think about what it is going to be doing, and what values it will take on. If we think about landing, it computes a bunch of different quantities, but really only has a small subset variables that we track across the trajectory. These are the ones that change with time throughout the entire mission, and likewise do so in landing too.

Mass - initial comes from previous phase; final is computed (though I don't see this in landing, perhaps very little fuel is burned.
Velocity - initial comes from previous phase; final is 0.
Distance - initial comes from previous phase, final is computed by summing several of the computed distances like flare or decision.
Probably altitude - initial comes from previous phase, final is airport altitude.
Possibly throttle

Additionally, for each of those variables, there are only certain values for num_nodes that make sense. We could do 2 (start end), or 3 (start, touchdown, end), since we compute values for those, or maybe 4 (start, flare, touchdown, end).  You should look through the calculations and figure out what level of time-discretization makes the most sense, and then require num_nodes to be fixed to that amount. Any other value would break this, because it is not inherently vectorizable in time.

Also, most of the variables you vectorized should not be vectorized. Airport altitude is not changing as a function of time. What you did would work in a multi-mission or a fleet context, but we need discretization along the mission.

Reply:
Now, I rolled back the vectorization in landing EOM.